### DISCUSSION TITLE
How to add data to the various plotting figures?

### DISCUSSION THREAD

**Discussion by jackjansen** [colour-science/colour#751] [Q&A] [answered]
The documentation is extensive on all the various plot types, but I cannot find any information on how to combine data into a single plot. I have also looked at the examples, but either I miss something or I am looking in the wrong place.

I have a number of color measurements (originally in a known RGB space, but I can convert these to xy or uv or another space). I want to plot these on a chromaticity diagram, possibly with additionally a CCT curve (`plot_planckian_locus()`) and a   colourspace (`plot_RGB_colourspaces_in_chromaticity_diagram()`).

1. I cannot find information on how to plot individual datapoints in a chromaticity diagram (except for the `plot_sds_` methods, but these are specifically for spectral distributions).
2. I cannot find how to combine graphs, so for example create a plot with both a CCT curve and a colourspace.

And while my current problem at hand is with chromaticity diagrams the same appears to be true for the other graph types I have looked at: the toplevel methods in `colour.plotting` all use some methods from sub packages such as `colour.plotting.temperature` or `colour.plotting.models`, and it is possible to override these methods from the sub packages but I cannot find any information on combining them.

**Comment by sobotka** [comment]
> 1. I cannot find information on how to plot individual datapoints in a chromaticity diagram (except for the plot_sds_ methods, but these are specifically for spectral distributions).

https://colour.readthedocs.io/en/latest/generated/colour.plotting.plot_RGB_chromaticities_in_chromaticity_diagram_CIE1931.html#colour.plotting.plot_RGB_chromaticities_in_chromaticity_diagram_CIE1931

It will follow the same format as `matplotlib` if you can't find the functionality with the default functions. For example, [you can plot the CIE 1931 chromaticity diagram](https://colour.readthedocs.io/en/latest/generated/colour.plotting.plot_chromaticity_diagram_CIE1931.html#colour.plotting.plot_chromaticity_diagram_CIE1931), and then add plot points with the array of x and y values. `matplotlib.pyplot.plot(x_array, y_array, [...])` should do the trick.

> 2. I cannot find how to combine graphs, so for example create a plot with both a CCT curve and a colourspace.

At this point you'll probably want to set `standalone=False` [in your generic plot using one of the typical ones from the library](https://colour.readthedocs.io/en/latest/colour.plotting.html), and then manually plot your run of CCT and then the colourspace triangle of the footprint of the prism. Using the generic `matplotlib` functionality is about the best way I've found to "combine" different plot points. I'm sure @KelSolaar might have some further tips.

Good luck!

**Comment by jackjansen** [comment]
Thank you very much, that should get me underway!

I still have the feeling that it would be good if this was documented, there is very little documentation (or example code) for adapting graphs. For example, I discovered the `standalone=False` option only by browsing the source code, because it is documented in the `render` class, not on the toplevel `plotting` methods, which makes it difficult to find.

I think an example that shows how to get the same effect as one of the standard plots by using `standalone=False` and adding things, manually calling `plot.show()`, etc would be helpful. I'll try to come up with something in the coming weeks.

**Comment by sobotka** [comment]
Feel free to update this report if you need further help!

The `standalone=False` is an extension of MatPlotLib arguments, which is a deep rabbit hole unto itself. Given Colour builds atop MatPlotLib, it is what it is, and fair to accept that to maximize plot control that it will require a bit of MatPlotLib research.

**Comment by thomasmansencal** [comment, marked as answer]
Hello @jackjansen,

There is an example on how to do that at the end of the tutorial:

- https://colour.readthedocs.io/en/develop/tutorial.html#convert-to-chromaticity-coordinates
- https://colab.research.google.com/drive/1Im9J7or9qyClQCv5sPHmKdyiQbG4898K#scrollTo=2uMiT_4bqdOT

or here in the How-To guide:

- https://colab.research.google.com/drive/1NRcdXSCshivkwoU2nieCvC3y14fx1X4X#scrollTo=UOFT98Gjd1De

We have also some of the most advanced examples (although they will probably not run with the latest develop) in the Cinematic Color 2 Figures notebook:

- https://colab.research.google.com/drive/1bmVU8fI1Rv3GLXK8kVDovc2K2roGjW9G?usp=sharing

I have been meant to expand add some more stuff to the documentation about the various plotting tips'n'tricks.

Hope that helps,

Cheers,

Thomas

**Comment by jackjansen** [comment]
Ah! https://colour.readthedocs.io/en/develop/tutorial.html#convert-to-chromaticity-coordinates is _exactly_ the example I would have needed. But, being the last example I didn't get this far, also because its title doesn't suggest that it has anything to do with plotting.

Can I suggest you modify the title of the example to something like _Converting to chromaticy coordinates and advanced plotting functionality_?

**Comment by thomasmansencal** [comment]
I think more than that we need a good dedicated section in the docs we can refer to and highlighting how things works! But yeah, we will make that more obvious as I realise that while I'm comfortable with the plotting API, many people are not.

**Comment by thomasmansencal** [comment]
Closing this one for now! Feel free to add comments and I will re-open!
